### PR TITLE
Added Redis for Dummies from Redis site

### PR DESCRIPTION
**Pull request by mfvitale**
## What does this PR do?
Add resource

### Description
- Added Redis for Dummies book from Redis official site

### Why is this valuable (or not)?
- Valuable for beginners

### How do we know it's really free?
It's available from Redis site leaving a mail address, Then there will be a download link

### For book lists, is it a book? For course lists, is it a course? etc.
- book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by davorpa** [comment]
Recover

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1511,6 +1511,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [MongoDB Koans](https://github.com/chicagoruby/MongoDB_Koans)
 * [MongoDB Succinctly, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/mongodb) (PDF, Kindle) (email address *requested*, not required)
 * [NoSQL Databases](http://www.christof-strauch.de/nosqldbs.pdf) - Christof Strauch (PDF)
+* [Redis-For-Dummies](https://redis.com/redis-for-dummies/) (PDF) *(email address or account required)*
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1514]:
```suggestion
* [Redis for Dummies](https://redis.com/redis-for-dummies/) - Redis Ltd ([PDF](https://redis.com/wp-content/uploads/2021/09/Redis-For-Dummies_FINAL.pdf)) *(email address or account required)*
```

**Comment by mfvitale** [comment]
I opened by mistake before hacktoberfest. I will open another tomorrow.

**Review by davorpa — CHANGES_REQUESTED** [review]

**Comment by davorpa** [comment]
> I opened by mistake before hacktoberfest. I will open another tomorrow.

If is your preference, **close it again**.  But I do not see much sense.

Anyways, you can integrate the review suggestion meanwhile :wink:

**Comment by davorpa** [comment]
Resubmitted at #5625

**Pull request closed without merging**